src/utils.jl

```diff
@@ -77,6 +77,55 @@ function sideof(p::Point{2,T}, c::Chain{2,T}) where {T}
   ifelse(isapprox(w, zero(T), atol=atol(T)), :OUTSIDE, :INSIDE)
 end
 
+"""
+    sideof(point, mesh)
+
+Determines whether a `point` is inside, outside or on the surface of a `mesh`.
+Possible results are `:INSIDE`, `:OUTSIDE`, or `:ON`.
+
+### Notes
+
+Uses a ray-casting algorithm.
+"""
+function sideof(point::Point{3,T}, mesh::Mesh{3,T}) where {T}
+  if !(eltype(mesh) <: Triangle)
+    throw(ArgumentError(
+          "This function only works for surface meshes with triangles as elements."))
+  end
+
+  z = last.(coordinates.(extrema(mesh)))
+  r = Ray(point, Vec(zero(T), zero(T), 2 * (z[2] - z[1])))
+
+  intersects = false
+  edgecrosses = 0
+  ps = Point{3,T}[]
+  for t in mesh
+    result = intersection(r, t) do I
+      if type(I) == CrossingRayTriangle
+        intersects = !intersects
+      elseif type(I) ∈ (EdgeTouchingRayTriangle,
+                        CornerTouchingRayTriangle,
+                        TouchingRayTriangle)
+        return :ON
```

Comment:
what is the effect of the return statement here, is it breaking the do block or the entire for loop?

Reply:
Do we actuallly gain type instability here using the 3-arg version of the intersection function as originally proposed? If we don't get type stability, I think we can just revert back to the 2-arg version to make the code more readable?

Reply:
The 2 arg is type stable. The reason for using the 3 arg is compiler optimizations as you mentioned before. Given my comments that this algorithm's advantage is robustness and we should implement an approximate algorithm for the users that need speed, I think the 2 arg is fine to use here.

Reply:
Yes, I think it is wise to stick to the 2-arg version here, at least in this first implementation. We can then come back to it later after it is merged and propose small improvements.

Reply:
I typically prefer that development style anyways (avoiding premature optimizations).